crates/polars-core/src/chunked_array/metadata/env.rs

```diff
@@ -0,0 +1,119 @@
+#[derive(Debug, Clone, Copy)]
+pub struct MetadataEnv(#[cfg(debug_assertions)] u32);
+
+#[cfg(debug_assertions)]
+impl MetadataEnv {
+    pub const ENABLED: u32 = 0x1;
+    pub const EXTENSIVE_USE: u32 = 0x2;
+    pub const LOG: u32 = 0x4;
+
+    #[inline(always)]
+    fn get_cached() -> Self {
+        static CACHED: std::sync::OnceLock<MetadataEnv> = std::sync::OnceLock::new();
+        *CACHED.get_or_init(Self::get)
+    }
+
+    #[inline(always)]
+    fn get() -> Self {
+        let Ok(env) = std::env::var("POLARS_METADATA_FLAGS") else {
+            return Self(Self::ENABLED);
+        };
+
+        if env == "0" {
+            return Self(0);
+        }
+
+        // @NOTE
+        // We use a RwLock here so that we can mutate it for specific runs or sections of runs when
```

Comment:
Will we ever do this in release? This is merely for debugging I assume? Is this CACHED even set in release?

Reply:
Nothing until line 80 is ever ran in release mode. We don't even cache anything in release mode, we just say `true` or `false`.